lib/Parse/ParseExpr.cpp

```diff
@@ -2966,7 +2966,7 @@ ParserStatus Parser::parseExprList(tok leftTok, tok rightTok,
     Expr *SubExpr = nullptr;
     if (Tok.isBinaryOperator() && peekToken().isAny(rightTok, tok::comma)) {
       SyntaxParsingContext operatorContext(SyntaxContext,
-                                           SyntaxContextKind::Expr);
+                                           SyntaxKind::BinaryOperatorExpr);
```

Comment:
This isn't necessarily a `BinaryOperatorExpr`.  Consider:

```swift
prefix operator ++++
prefix func ++++ (x: Int) -> Int { return x + 4 }
let plus4: (Int) -> Int = (++++)
print(plus4(12))
```
Could you make this `IdentifierExpr`, and make `IdentifierExpr` to accept `*OperatorToken`?

Reply:
Interestingly, it *is* always parsed as a binary operator. Even in 
```swift
prefix operator ++++
prefix func ++++ (x: Int) -> Int { return x + 4 }
func foo(x: (Int) -> Int) -> Int {
  return x(4)
}
print(foo(x: ++++))
```
In the last line the `++++` is parsed as a `BinaryOperatorExpr`. Looking into it, it seems like the lexer treats all operators as binary operator tokens to start with. I guess that changing this will be rather difficult and far-reaching, so I suggest keeping it as it is even though it's not technically correct.

Reply:
I know it's tokenized as binary operator *token* and I'm OK with it.
But still, I don't like it's parsed as binary operator *expression*. `BinaryOperatorExpr` in `libSyntax` should appear only in `SequenceExpr`.

That being said, I don't have strong opinion on it.
If you decide to keep this `BinaryOperatorExpr`, please update this comment. https://github.com/apple/swift/blob/f6b67f33/utils/gyb_syntax_support/ExprNodes.py#L160